kotlin/samples/containers/hello-back-button/src/main/java/com/squareup/sample/hellobackbutton/AreYouSureWorkflow.kt

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright 2020 Square Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package com.squareup.sample.hellobackbutton
+
+import com.squareup.sample.hellobackbutton.AreYouSureWorkflow.Finished
+import com.squareup.sample.hellobackbutton.AreYouSureWorkflow.State
+import com.squareup.sample.hellobackbutton.AreYouSureWorkflow.State.Quitting
+import com.squareup.sample.hellobackbutton.AreYouSureWorkflow.State.Running
+import com.squareup.sample.container.BackButtonScreen
+import com.squareup.workflow.RenderContext
+import com.squareup.workflow.Snapshot
+import com.squareup.workflow.StatefulWorkflow
+import com.squareup.workflow.WorkflowAction.Companion.noAction
+import com.squareup.workflow.action
+import com.squareup.workflow.ui.AlertContainerScreen
+import com.squareup.workflow.ui.AlertScreen
+import com.squareup.workflow.ui.AlertScreen.Button.NEGATIVE
+import com.squareup.workflow.ui.AlertScreen.Button.POSITIVE
+import com.squareup.workflow.ui.AlertScreen.Event.ButtonClicked
+import com.squareup.workflow.ui.AlertScreen.Event.Canceled
+
+/**
+ * Wraps [HelloBackButtonWorkflow] to (sometimes) pop a confirmation dialog when the back
+ * button is pressed.
+ */
+object AreYouSureWorkflow : StatefulWorkflow<Unit, State, Finished, AlertContainerScreen<*>>() {
+  override fun initialState(
+    props: Unit,
+    snapshot: Snapshot?
+  ): State = Running
+
+  enum class State {
+    Running,
+    Quitting
+  }
+
+  object Finished
+
+  override fun render(
+    props: Unit,
+    state: State,
+    context: RenderContext<State, Finished>
+  ): AlertContainerScreen<*> {
+    val ableBakerCharlie = context.renderChild(
+        HelloBackButtonWorkflow, Unit) { noAction() }
+
+    return when (state) {
+      Running -> {
+        AlertContainerScreen(
+            BackButtonScreen(ableBakerCharlie) {
+              // While we always provide a back button handler, by default the view code
+              // associated with BackButtonScreen ignores ours if the view created for the
+              // wrapped rendering sets a handler its own. (Set BackButtonScreen.override
```

Comment:
`…of its own…`

Reply:
Well shit. I just realized that this override check code is inadequate because it doesn't recurse.

Reply:
But I'm pretty sure I can get he same effect by using Android's `OnBackPressedDispatcherOwner` (the underpinnings of our `View.backPressedHandler`) correctly.